src/eredis_cluster_manager.erl

```diff
@@ -0,0 +1,415 @@
+get_key_from_rest([_,KeyName|_]) when is_list(KeyName) ->
+    KeyName;
+get_key_from_rest(_) ->
+    undefined.
+
+%% =============================================================================
+%% @doc Return the hash slot from the key
+%% @end
+%% =============================================================================
+-spec get_key_slot(Key::any()) -> Slot::integer().
+get_key_slot(K) ->
+    %% cast any type of key to a list.
+    Key = lists:concat([K]),
+    KeyToBeHashed = case string:chr(Key,${) of
+                        0 ->
+                            Key;
+                        Start ->
+                            case string:chr(string:substr(Key,Start+1),$}) of
+                                0 ->
+                                    Key;
+                                Length ->
+                                    if
+                                        Length =:= 1 ->
+                                            Key;
+                                        true ->
+                                            string:substr(Key,Start+1,Length-1)
+                                    end
+                            end
+                    end,
+    eredis_cluster_hash:hash(KeyToBeHashed).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Methods to manage list of eredis connections
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+init_eredis_connections() ->
+    ets:new(?MODULE, [protected, set, named_table, {read_concurrency, true}]).
+
+%% Looks up existing redis client pid using Ip, Port and return Pid of the client
+%% connection. If no such connection is present, this method establishes the connection.
+-spec lookup_eredis_pid(Ip::string(), Port::integer()) -> {ok, Pid::pid()}.
+lookup_eredis_pid(Ip, Port) ->
+    case ets:lookup(?MODULE, [Ip, Port]) of
+        [] ->
+           {ok, Pid} = safe_eredis_start_link(Ip, Port),
+           ets:insert(?MODULE, {[Ip, Port], Pid}),
+          {ok, Pid};
+        [_, Pid] -> {ok, Pid}
+    end.
+
+
+%% Closes redis client connection if present in cstate.
+%%-spec delete(Ip::string(), Port::integer()) -> {ok}.
+%%delete(Ip, Port) ->
+%%    case ets:lookup(?MODULE, [Ip, Port]) of
+%%        [_, Pid] ->
+%%             safe_eredis_stop(Pid),
+%%             ets:remove(?MODULE, [Ip, Port]),
+%%            ok; 
+%%        {_, _} -> {ok}
+%%    end. 
+
+
+safe_eredis_start_link(Ip, Port) ->
+    process_flag(trap_exit, true),
+    Payload = eredis:start_link(Ip, Port),
+    process_flag(trap_exit, false),
+    Payload.
+
+
+%%safe_eredis_stop(Pid) ->
+%%    process_flag(trap_exit, true),
+%%    eredis:stop(Pid),
+%%    process_flag(trap_exit, false).
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% gen_server call backs
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+init([Args]) ->
+    init_eredis_connections(),
```

Comment:
should be renamed to create_ets_connection_table

Reply:
I think done.